public/app/features/dashboard-scene/scene/layout-responsive-grid/ResponsiveGridItemRenderer.tsx

```diff
@@ -3,22 +3,34 @@ import { css, cx } from '@emotion/css';
 import { SceneComponentProps } from '@grafana/scenes';
 import { useStyles2 } from '@grafana/ui';
 
+import { useDashboardState, useIsConditionallyHidden } from '../../utils/utils';
+
 import { ResponsiveGridItem } from './ResponsiveGridItem';
 
 export function ResponsiveGridItemRenderer({ model }: SceneComponentProps<ResponsiveGridItem>) {
   const { body } = model.useState();
   const style = useStyles2(getStyles);
+  const { showHiddenElements } = useDashboardState(model);
+  const isConditionallyHidden = useIsConditionallyHidden(model);
+
+  if (isConditionallyHidden && !showHiddenElements) {
```

Comment:
We can avoid some unnecessary re-renders if we move the conditionally hidden check directly in the `Component` property